mlflow/sklearn/__init__.py

```diff
@@ -1262,15 +1263,25 @@ def fit_mlflow_xgboost_and_lightgbm(original, self, *args, **kwargs):
         """
         Autologging function for XGBoost and LightGBM scikit-learn models
         """
+        # Obtain a copy of a model input example from the training dataset prior to model training
+        # for subsequent use during model logging, ensuring that the input example and inferred
+        # model signature to not include any mutations from model training
+        input_example = _get_X_y_and_sample_weight(self.fit, args, kwargs)[0][
+            :INPUT_EXAMPLE_SAMPLE_ROWS
+        ]
         # parameter, metric, and non-model artifact logging are done in
         # `train()` in `mlflow.xgboost.autolog()` and `mlflow.lightgbm.autolog()`
         fit_output = original(self, *args, **kwargs)
         # log models after training
-        X = _get_X_y_and_sample_weight(self.fit, args, kwargs)[0]
```

Comment:
By this time, `X` has already been mutated by any sckit-learn transformations defined by the estimator.